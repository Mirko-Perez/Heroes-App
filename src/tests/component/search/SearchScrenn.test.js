import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { SearchScreen } from "../../../components/search/SearchScrenn";

const mockNavigate=jest.fn()


jest.mock('react-router-dom',()=>({
  ...jest.requireActual('react-router-dom'),
  useNavigate:()=>mockNavigate
}));


describe('Pruebas en el componente <SearchScrenn />', () => {
  
    test('Debe de mostrarse correctamente con valores por defecto', () => {
      
      const wrapper = mount(
      
      <MemoryRouter initialEntries={['/search']}>
        <SearchScreen />
      </MemoryRouter>
      
      );
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('.alert-info').text().trim()).toBe('Buscar un HÃ©roe')

    });

    test('Debe de mostrar a Batman y el input con el valor queryString', () => {

      const wrapper = mount(
      
        <MemoryRouter initialEntries={['/search?q=batman']}>
          <SearchScreen />
        </MemoryRouter>
      );
      expect(wrapper.find('input').prop('value')).toBe('batman')

      
    });

    test('Debe de mostrar un error si el query no existe', () => {
      
      const wrapper = mount(
      
        <MemoryRouter initialEntries={['/search?q=batman123']}>
          <SearchScreen />
        </MemoryRouter>
      );
      expect(wrapper.find('.alert-danger').text().trim()).toBe('No hay resultados con: batman123')

    });
    
    test('Debe de llamar el navigate a el nuevo url', () => {
      
      const wrapper = mount(
      
        <MemoryRouter initialEntries={['/search']}>
          <SearchScreen />
        </MemoryRouter>
      );
      
      wrapper.find('input').simulate('change',{
        target:{
          name:'searchText',
          value:'Batman'
        }

      })

      wrapper.find('form').prop('onSubmit')({
        preventDefault(){}
      })

      expect(mockNavigate).toHaveBeenCalled()


    });
    
    
    
});

