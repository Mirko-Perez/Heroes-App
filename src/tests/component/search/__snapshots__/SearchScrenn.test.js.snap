// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <SearchScrenn /> Debe de mostrarse correctamente con valores por defecto 1`] = `
Array [
  <h1>
    Searches
  </h1>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Search for a hero"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-primary mt-1"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Buscar un HÃ©roe
      </div>
    </div>
  </div>,
]
`;
